N/NetCDF/build_tarballs.jl

```diff
@@ -20,31 +20,29 @@ version = VersionNumber(upstream_version.major * 100 + version_offset.major,
 # Collection of sources required to build NetCDF
 sources = [
     ArchiveSource("https://downloads.unidata.ucar.edu/netcdf-c/$(upstream_version)/netcdf-c-$(upstream_version).tar.gz",
-                  "38f62cc5d9f0409205b20ff8ff01d7cc36659993"),
+                  "cf11babbbdb9963f09f55079e0b019f6d0371f52f8e1264a5ba8e9fdab1a6c48"),
 ]
 
 # HDF5.h in /workspace/artifacts/805ccba77cd286c1afc127d1e45aae324b507973/include
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/netcdf-c
+cd $WORKSPACE/srcdir/netcdf-c*
 
 export CPPFLAGS="-I${includedir}"
 export LDFLAGS="-L${libdir}"
 export LDFLAGS_MAKE="${LDFLAGS}"
 CONFIGURE_OPTIONS=""
 
 if [[ ${target} == *-mingw* ]]; then
-    export LIBS="-lhdf5-0 -lhdf5_hl-0 -lcurl-4 -lz"
+    # we should determine the dll version (?) automatically
```

Comment:
Yeah, to be honest I'm a bit worried about those random numbers like 310.  Where they come from?  Will they change in hdf5 1.15?  If so, it'd be ABI-incompatible with v1.14

Reply:
I think that's essentially the [soversion](https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html) of the library, produced by HDF5's build system. We can probably, if we want to, modify the Windows build script for HDF5 to change these numbers to `0`.

On the other hand, if these numbers indicate binary (ABI) compatibility, then we should probably heed them, and be quite strict about the `compat` declaration when using `HDF5_jll`.

Reply:
> On the other hand, if these numbers indicate binary (ABI) compatibility

Yes, the SONAME is part of the ABI, whenever it changes the ABI changes.  If they change the soname for every minor version (sigh) we really must be strict about compat bounds.